docs/rfc/active/business_glossary/README.md

```diff
@@ -0,0 +1,224 @@
+
+## Detailed design
+
+### What is Business Glossary 
+**Business Glossary**, is a list of business terms with their definitions. It defines business concepts for an organization or industry and is independent from any specific database or platform or vendor.
+
+**Data Dictionary** is a description of a data set, provides the details about the attributes and data types
+
+### Relationship 
+Even though Data Dictionary and Business Glossary are separate entities, they work nicely together to describe different aspects and levels of abstraction of the data environment of an organization.
+Business terms can be linked to specific entities/tables and columns in a data asset/data dictionary to provide more context and consistent approved definition to different instances of the terms in different platforms/databases.
+
+
+### Sample Business Glossary Definition
+|URN|Business Term |Definition  | Domain/Namespace | Owner | Ext Source| URI | Ext Reference |
+|--|--|--|--|--|--|--|--|
+|urn:li:businessTerm:(instrument.cashInstrument) | instrument.cashInstrument| time point including a date and a time, optionally including a time zone offset| Foundation | abc@domain.com | fibo | fibo-fbc-fi-fi:CashInstrument|https://spec.edmcouncil.org/fibo/ontology/FBC/FinancialInstruments/FinancialInstruments/CashInstrument |
+|urn:li:businessTerm:(common.dateTime) | common.dateTime| a financial instrument whose value is determined by the market and that is readily transferable (highly liquid)| Finance | xyz@domain.com | fibo | fibo-fnd-dt-fd:DateTime|https://spec.edmcouncil.org/fibo/ontology/FND/DatesAndTimes/FinancialDates/DateTime |
+|urn:li:businessTerm:(market.bidSize) | market.bidSize| The bid size represents the quantity of a security that investors are willing to purchase at a specified bid price| Trading | xyz@domain.com | - | - |- | - |
+|--|--|--|--|--|--|--|--|
+| | | | | | | | |
+
+### Business Term  & Dataset - Relationship
+
+| Attribute Name| Data Type| Nullable?| **Business Term**| Description|
+|--|--|--|--|--|
+| recordId| int| N| | |n the case of FX QuoteData the RecordId is equal to the UIC from SymbolsBase|
+| arrivalTime| TimestampTicks| N| | Time the price book was received by the TickCollector. 100s of Nanoseconds since 1st January 1970 (ticks)|
+| bid1Price| com.xxxx.yyy.schema.common.Price| N| **common.monetoryAmount**|The bid price with rank 1/29.|
+| bid1Size| int| N| market.bidSize| The amount the bid price with rank 5/29 is good for.|
+|--|--|--|--|--|--|--|
+| | | | | | | |
+
+### Stiching Together
+
+
+Business Glossary will be a first class entity where one can define the business terms and this will be similar to entities like Dataset, CorporateUser etc.
+Then extending the Dataset's aspect SchemaMetadata, that will be extending the SchemaField model to capture urn of the business term (optional).
```

Comment:
On thinking more ...  we should make dataset field as a first class entity. We might want to capture more and more metadata of a dataset field  ( business glossary relation , documentations , annotations deprecated flag  etc..,)

In this RFC https://github.com/linkedin/datahub/pull/1841   .. I was actually proposing an URN for dataset field and providing path for making it a first class entity. 

WDYT ?

Reply:
listing down on what it means to make dataset field first class entity 
- Define aspects 
- More data in KV store ( graph db will anyhow be added with fields after my RFC and implementation. So, no change in graph db storage) 
- Define resources


First class entity requires URN as a unique identifier. So, once we define URN ... the next steps is define aspects and code the resources (APIs) serving the entity information.

Reply:
Thanks Madhu for your agreement on having DatasetFieldUrn and making it a first class entity. 

I updated my PR  #1841 discussing more on the dataflow once we make dataset field a first class entity. You might want to update your RFC in those similar lines ?

Reply:
I believe I should reference your RFC(#[1841](https://github.com/linkedin/datahub/pull/1841)), once that is made to master branch, your thoughts? if you still say, update this **`rfc`**, i shall do the same

Reply:
I would suggest to update the RFC referring to DatasetField entity instead of expanding SchemaField.pdl. So, we don't need to update this RFC again.

Reply:
make sense
Let me reference the DatasetField rather than the SchemaField.pdl, I could not find the schema definition for the DatasetField entity in your RFC (may point to the spec of DatasetField entity)

Reply:
Hi Nagarjuna, have incorporated the feedback and added the relationship as well